File: primerTrimestre/introduccion_programacion_python/3repetitivas_py/ej10rep.py
# Programa: ej10rep.py
# Propósito: Pide una cadena y un carácter por teclado y muestra cuantas
# veces aparece el carácter en la cadena.
#
# Autor: Antonio Quesada
# Fecha: 30/10/2019.
#
# Variables a usar:
#   * caracter
#   * cadena

#Pedimos que introduzca una cadena.
cadena = str(input("Introduce una cadena:"))

#Bucle while.
while True:
    
    #Introduce el caracter que debe contar en la cadena.
    caracter = input("Introduce un carácter:")
    
    if len(caracter)==1: break
    
    #Inicializamos el contador a 0.
cont = 0

    #Bucle for. Usamos la funcion len que devuelve la longitud de la cadena de caracter.
for posicion in range(len(cadena)):
    
    #Realizo una lista para que en la cadena encuentre el caracter.
    if cadena[posicion-1:posicion]==caracter:
        
        #Contador para el número de veces que se repite el carácter.
        cont = cont+1
print("Aparece",cont," veces el carácter ",caracter)
    

File: primerTrimestre/java basico/3repetitivas/Ej10rep.java
/**Pide una cadena y un carácter por teclado y muestra cuantas veces aparece el carácter en la cadena.
 *
 * @author Antonio Quesada
 */

import java.util.Scanner;

public class Ej10rep {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    String cadena;
    String caracter;
    
    //Pedimos una cadena y un caracter.
    System.out.println("Introduce una cadena: ");
    cadena = s.nextLine();
    System.out.println("Introduce un caracter: ");
    caracter = s.nextLine();
   
    //Inicializamos el contador a 0.
    int contador = 0;
    
    //Bucle for. Devuelve la longitud de la cadena.
    //Compara la posicion del caracter en la cadena 
    for (int i = 0; i < cadena.length(); i++){
      if (cadena.charAt(i)==caracter.charAt(0)){
          contador++; //Contador para el número de veces que se repite el caracter.
          System.out.println("El carácter se repite " + contador + " veces");
      }
      
      //Si no se encuentra el caracter en la cadena lo muestra.
      else {
    } System.out.println("El caracter introducido no se encuentra en la cadena");
      break;
    }
      
  }

}
